### DISCUSSION TITLE
Do I need to do anything to transform exceptions' error messages?

### DISCUSSION THREAD

**Discussion by apreifsteck** [open-telemetry/opentelemetry-erlang#716] [Q&A]
Hello!

I recently experimented with adding open telemetry to one of the apps my team owns. It's been over a week now since I made the swap and we've had a few exceptions. However, they all show up in DataDog with no stacktrace info and this message: `exit:{{#{'__exception__' => true,...},[...]},{'Elixir.MyAppWeb.Endpoint',...}}`. Do I need to add some transformation logic? I haven't seen this mentioned anywhere else really, so I'm wondering if maybe I misconfigured something. I would appreciate any help!

**Comment by tsloughter** [comment]
Are you calling `record_exception` manually or using an instrumentation library?

You've swapped out Spandex maybe? Did those exceptions show up right in datadog?

**Reply by apreifsteck** [reply]
Using instrumentation libraries. I was assuming that any exceptions that occurred after the router would show up with correct error info in the `opentelemetry_cowboy.server` traces (I don't see anything tagged for phoenix, but maybe that's how it's supposed to be?)

And you're exactly correct, we swapped from Spandex. Those exceptions were showing up correctly in datadog.

**Reply by tsloughter** [reply]
Good to know this is an area we are lacking in parity with Spandex.

Can you open an issue in https://github.com/open-telemetry/opentelemetry-erlang-contrib

**Reply by apreifsteck** [reply]
https://github.com/open-telemetry/opentelemetry-erlang-contrib/issues/306
